### PR TITLE
Update PythonMap transform to accept extra packages

### LINKED ISSUE #25221: [Feature Request]: Update PythonMap transform to accept extra packages
### What would you like to happen?

This will allow executing PythonMap [1] x-lang transform with functions that require external dependencies.

[1] https://github.com/apache/beam/blob/master/sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/PythonMap.java

### Issue Priority

Priority: 2 (default / most feature requests should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by chamikaramj**
This fixes #25221 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by chamikaramj** [comment]
Run XVR_JavaUsingPython_Dataflow PostCommit

**Comment by chamikaramj** [comment]
Run XVR_JavaUsingPython_Dataflow PostCommit

**Comment by chamikaramj** [comment]
R: @ahmedabu98

**Comment by chamikaramj** [comment]
Run XVR_JavaUsingPython_Dataflow PostCommit

File: sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/PythonMap.java [hunk 1]
```diff
@@ -59,13 +63,34 @@ public PythonMap<InputT, OutputT> withExpansionService(String expansionService)
...
+   * {@link #withExpansionService(String)} to provide an expansion service.
+   *
+   * <p>For model handlers provided by Beam Python SDK, the implementation will automatically try to
+   * infer correct packages needed, so this may be omitted.
+   *
+   * @param extraPackages a list of PyPi packages. May include the version.
+   * @return A {@link PythonMap} with extra packages.
+   */
+  public PythonMap<InputT, OutputT> withExtraPackages(List<String> extraPackages) {
+    if (!this.extraPackages.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (chamikaramj) [reply sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/PythonMap.java:78]: We already perform such a check at PythonExternalTransform so don't think we need to repeat the check here.

The check is to make sure that the expansion service is not specified BTW. withExtraPackages only works with the auto-started expansion services.

https://github.com/apache/beam/blob/9fcb3a5b48a6db0dcf57f454d1d1eca10cf1c41b/sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/PythonExternalTransform.java#L292

File: sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/PythonMap.java [hunk 2]
```diff
@@ -59,13 +63,34 @@ public PythonMap<InputT, OutputT> withExpansionService(String expansionService)
...
   }
 
+  /**
+   * Specifies any extra Pypi packages required by the RunInference model handler.
+   *
+   * <p>This should only be specified when using the default expansion service, i.e. when not using
+   * {@link #withExpansionService(String)} to provide an expansion service.
+   *
+   * <p>For model handlers provided by Beam Python SDK, the implementation will automatically try to
+   * infer correct packages needed, so this may be omitted.
```
[the remark this answers is not in the document]
Comment (chamikaramj) [reply sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/transforms/PythonMap.java]: Good catch. Done.

**Review by chamikaramj — COMMENTED** [review]
Thanks. PTAL.

I had to remove the IT since it ended up running with a manual expansion service which is incompatible with this change.

**Comment by ahmedabu98** [comment]
Run Java PreCommit

**Comment by ahmedabu98** [comment]
Run XVR_JavaUsingPython_Dataflow PostCommit

**Review by ahmedabu98 — APPROVED** [review]
LGTM

**Comment by chamikaramj** [comment]
Thanks!

**Pull request merged**